File: backend/catalog-service/src/main/java/me/doruk/catalogservice/service/VenueService.java
package me.doruk.catalogservice.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import me.doruk.catalogservice.entity.Venue;
import me.doruk.catalogservice.repository.VenueRepository;
import me.doruk.catalogservice.request.VenueCreateRequest;
import me.doruk.catalogservice.response.VenueResponse;

import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

import java.util.List;

@Service
@Slf4j
@RequiredArgsConstructor
public class VenueService {

  private final VenueRepository venueRepository;

  public VenueResponse getVenueInformation(final Long venueId) {
    System.out.println("Fetching venue information for venueId: " + venueId);
    final Venue venue = venueRepository.findById(venueId)
        .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Venue not found"));

    return VenueResponse.builder()
        .id(venue.getId())
        .name(venue.getName())
        .location(venue.getLocation())
        .totalCapacity(venue.getTotalCapacity())
        .imageUrl(venue.getImageUrl())
        .build();
  }

  public List<VenueResponse> getAllVenues() {
    final List<Venue> venues = venueRepository.findAll();

    return venues.stream().map(venue -> VenueResponse.builder()
        .id(venue.getId())
        .name(venue.getName())
        .location(venue.getLocation())
        .totalCapacity(venue.getTotalCapacity())
        .imageUrl(venue.getImageUrl())
        .build()).toList();
  }

  public VenueResponse createVenue(final VenueCreateRequest request) {
    System.out.println("Creating venue: " + request);
    Venue venue = new Venue();
    venue.setName(request.getName());
    venue.setLocation(request.getLocation());
    venue.setTotalCapacity(request.getTotalCapacity());

    Venue savedVenue = venueRepository.save(venue);

    return VenueResponse.builder()
        .id(savedVenue.getId())
        .name(savedVenue.getName())
        .location(savedVenue.getLocation())
        .totalCapacity(savedVenue.getTotalCapacity())
        .imageUrl(savedVenue.getImageUrl())
        .build();
  }

}

File: backend/CatalogService/Services/VenueService.cs
using Microsoft.EntityFrameworkCore;

using CatalogService.Entities;
using CatalogService.Repositories;
using CatalogService.Requests;
using CatalogService.Responses;

namespace CatalogService.Services;

public class VenueService
{
    private readonly IVenueRepository _venueRepository;

    public VenueService(IVenueRepository venueRepository)
    {
        _venueRepository = venueRepository;
    }

    public async Task<List<VenueResponse>> GetAllVenues()
    {
        Console.WriteLine("GetAllVenues");

        var venues = await _venueRepository.GetAllVenuesAsync();

        return venues
            .Select(v => new VenueResponse(
                v.Id,
                v.Name,
                v.Location,
                v.TotalCapacity,
                v.ImageUrl
            ))
            .ToList();
    }

    public async Task<VenueResponse> GetVenueInformation(long venueId)
    {
        Console.WriteLine($"Fetching venue information for venueId: {venueId}");

        var venue = await _venueRepository.GetVenueAsync(venueId)
            ?? throw new KeyNotFoundException("Venue not found");

        return new VenueResponse(
                venue.Id,
                venue.Name,
                venue.Location,
                venue.TotalCapacity,
                venue.ImageUrl
            );
    }

    public async Task<VenueResponse> CreateVenue(VenueCreateRequest request)
    {
        Console.WriteLine($"Creating venue: {request}");

        var venue = new Venue(
            id: 0,
            name: request.Name,
            location: request.Location,
            totalCapacity: request.TotalCapacity,
            imageUrl: string.IsNullOrEmpty(request.ImageUrl)
              ? "default-venue.jpg" : request.ImageUrl
        );

        await _venueRepository.AddVenueAsync(venue);

        return new VenueResponse(
            venue.Id,
            venue.Name,
            venue.Location,
            venue.TotalCapacity,
            venue.ImageUrl
        );
    }
}